src/plugins/intel_gpu/src/kernel_selector/kernel_base_opencl.cpp

```diff
@@ -177,11 +177,10 @@ std::shared_ptr<KernelString> KernelBaseOpenCL::GetKernelString(const std::strin
                 kernel_string->options += " -DOPT_HINTS_SUPPORTED=1";
         }
 
-#if CL_TARGET_OPENCL_VERSION >= 300
-        kernel_string->options += " -cl-std=CL3.0";
-#elif CL_TARGET_OPENCL_VERSION >= 200
-        kernel_string->options += " -cl-std=CL2.0";
-#endif
+        if (engine_info.supports_work_group)
+            kernel_string->options += " -cl-std=CL3.0";
+        else
+            kernel_string->options += " -cl-std=CL2.0";
```

Comment:
Why should the support/implementation of some feature affect the kernel's OpenCL version?

Reply:
[OpenCL_C::unified-spec](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_C.html#unified-spec)
the __opencl_c_<feature_name> feature: Features that were introduced in OpenCL C 2.0 as mandatory, but made optional in OpenCL C 3.0.
On Elkhart Lake(Legacy platform), OpenCL C 3.0 is working but work_group_<ops> APIs are not supported.
OpenCL C 2.0 includes work_group_<ops> on EHL.